dbms/src/Storages/DeltaMerge/Index/MinMaxIndex.cpp

```diff
@@ -99,6 +99,23 @@ ALWAYS_INLINE bool minIsNull(const DB::ColumnUInt8 & null_map, size_t i)
 {
     return null_map.getElement(i * 2);
 }
+
+ALWAYS_INLINE bool isLegacyOrHasNull(
+    const DB::ColumnUInt8 & null_map,
+    const PaddedPODArray<UInt8> & has_nulls,
+    size_t i)
+{
+    // Null is a special value. The result of comparing null with any value and the logical
+    // operation result of null value are unknown. So, if a pack has null, always return RSResult::Some.
+
+    // Why need return RSResult::Some when a pack has null?
+    // Take `MinMaxIndex::checkIn` for example.
+    // There is pack has data [1, 2, 3, 4, null] and a query `... not in (100)`.
```

Comment:
This will cause `... in (100, 101)` return `Some` instead of `None`, which may cause performance regression？